Spurious test failure seen in Travis

````
?   	_/home/travis/gopath/src/github.com/drawks/go-carbon	[no test files]
--- FAIL: TestCarbonlink-2 (0.05 seconds)
	Location:	carbonlink_test.go:115
	Error:		Not equal: "\x80\x02}(X\n\x00\x00\x00datapoints](J\xd3)\xceTG\xc0E\x11\xeb\x85\x1e\xb8R\x86J\xbe$\xceTG@.\x00\x00\x00\x00\x00\x00\x86eu." (expected)
			        != "\x80\x02}(X\n\x00\x00\x00datapoints](J\xd3)\xceTG\xc0E\x11\xeb\x85\x1e\xb8R\x86eu." (actual)
		
	Location:	carbonlink_test.go:139
	Error:		Not equal: "\x80\x02}(X\n\x00\x00\x00datapoints](eu." (expected)
			        != "\x80\x02}(X\n\x00\x00\x00datapoints](J\xbe$\xceTG@.\x00\x00\x00\x00\x00\x00\x86eu." (actual)
		
FAIL
FAIL	_/home/travis/gopath/src/github.com/drawks/go-carbon/cache	0.562s
````

While checking in something completely unrelated, I noticed this test error pop on travis for a go1.3 based build. Not sure if it may be related to #83 or not

--- a/cache/carbonlink_test.go
+++ b/cache/carbonlink_test.go
@@ -46,8 +46,11 @@ func TestCarbonlink(t *testing.T) {
 	cache.In() <- msg3
 
 	defer cache.Stop()
-
-	time.Sleep(50 * time.Millisecond)
+	for {
+		if len(cache.In()) == 0 {
+			break
+		}
+	}
 
 	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
 	assert.NoError(err)

